src/array.js

```diff
@@ -84,8 +84,12 @@ inherits(ArraySchema, MixedSchema, {
 
         originalValue = originalValue || value;
 
-        let validations = value.map((item, idx) => {
-          var path = makePath`${options.path}[${idx}]`;
+        let len = value.length;
+        let idx = -1;
+        let validations = new Array(len);
+        while (++idx < len) {
```

Comment:
i'd write this as a normal `for` loop it's also fine to not cache the length value as a separate variable

Reply:
Roger, will do!

I picked up this style from lodash's source code, but it's probably better to use the more readable `for` here.